resolver/resolver.cc

```diff
@@ -786,9 +838,21 @@ class ResolveConstantsWalk {
 
     ast::TreePtr postTransformSend(core::Context ctx, ast::TreePtr tree) {
         auto &send = ast::cast_tree_nonnull<ast::Send>(tree);
-        if (send.recv.isSelfReference() && send.fun == core::Names::mixesInClassMethods()) {
-            auto item = ClassMethodsResolutionItem{ctx.file, ctx.owner, &send};
-            this->todoClassMethods_.emplace_back(move(item));
+        if (send.recv.isSelfReference()) {
+            if (send.fun == core::Names::mixesInClassMethods()) {
+                auto item = ClassMethodsResolutionItem{ctx.file, ctx.owner, &send};
+                this->todoClassMethods_.emplace_back(move(item));
+            } else if (send.fun == core::Names::requiresAncestor()) {
+                if (!ctx.state.requiresAncestorEnabled) {
+                    if (auto e = ctx.beginError(send.loc, core::errors::Resolver::ExperimentalRequiredAncestor)) {
+                        e.setHeader("`{}` is an experimental feature. Use `{}` to enable it", "requires_ancestor",
+                                    "--enable-experimental-requires-ancestor");
+                    }
+                } else {
+                    auto item = RequireAncestorResolutionItem{ctx.file, ctx.owner, &send};
+                    this->todoRequiredAncestors_.emplace_back(move(item));
+                }
+            }
```

Comment:
Note that the method will still be defined for Sorbet from it's RBI even without the flag, so you will be calling it but it will `no-op`. It might be better to explicitly error

That said, I removed the error as you asked.